modules/test/playwright/tests/portal-web/html/taglib/ui/pageIterator.spec.ts

```diff
@@ -114,3 +126,151 @@ test(
+				name: 'Asset Selection',
+			});
+			await assetSelectionTab.waitFor({state: 'visible'});
+			await assetSelectionTab.click();
+
+			await clickAndExpectToBeVisible({
+				autoClick: true,
+
+				target: configurationIframe
+					.frameLocator('iframe[title="Select Collection"]')
+					.getByRole('link', {name: 'Collection Providers'}),
+				timeout: 2000,
+				trigger: configurationIframe.getByRole('button', {
+					exact: true,
+					name: 'Select Collection',
+				}),
+			});
+
+			await clickAndExpectToBeHidden({
+				target: configurationIframe.locator('.modal-dialog'),
+				timeout: 2000,
+				trigger: configurationIframe
+					.frameLocator('iframe[title="Select Collection"]')
+					.getByRole('button', {name: 'Select Recent Content'}),
+			});
+
+			await configurationIframe
+				.getByRole('tab', {name: 'Display Settings'})
+				.click();
+
+			const itemDisplayInput = configurationIframe.getByLabel(
+				'Number of Items to Display'
+			);
+
+			await itemDisplayInput.waitFor({state: 'visible'});
+
+			await itemDisplayInput.click();
+
+			await itemDisplayInput.fill('1');
+
+			await configurationIframe
+				.getByLabel('Pagination Type')
+				.selectOption('Regular');
+
+			await configurationIframe
+				.getByRole('button', {name: 'Save'})
+				.click();
+
+			await configurationIframe
+				.getByRole('button', {name: 'Cancel'})
+				.click();
+
+			await page.getByLabel('Publish', {exact: true}).click();
+		});
+
+		await test.step('Create web content articles and test dropdown', async () => {
+			await page.goto(
+				`/web${site.friendlyUrlPath}/${layout.friendlyUrlPath}`
+			);
+
+			for (let i = 1; i <= 10; i++) {
+				const contentStructureId =
+					await getBasicWebContentStructureId(apiHelpers);
+				const randomTitle = getRandomString();
+
+				const webContent =
+					await apiHelpers.jsonWebServicesJournal.addWebContent({
+						ddmStructureId: contentStructureId,
+						groupId: site.id,
+						titleMap: {en_US: randomTitle},
+					});
+
+				apiHelpers.data.push({
+					id: `${site.id}_${webContent.articleId}`,
+					type: 'webContent',
+				});
+			}
+
+			await page.reload();
+
+			await page.setViewportSize({height: 600, width: 300});
+
+			await page.evaluate(() => {
+
+				// @ts-ignore
+
+				document.body.style.zoom = '150%';
+
+				const intermediatePagesButton = document.querySelector(
+					'button[title="Show Intermediate Pages"]'
+				);
+
+				if (intermediatePagesButton) {
+					(intermediatePagesButton as HTMLElement).click();
+				}
+			});
+
+			const dropdownVisible = await page.evaluate(() => {
+				const dropdownMenu = document.querySelector(
+					'.dropdown-menu-top.dropdown-menu-width-shrink'
+				);
+
+				if (dropdownMenu) {
+					const dropdownMenuRect =
+						dropdownMenu.getBoundingClientRect();
+
+					const windowWidth = window.innerWidth;
+
+					return (
+						dropdownMenuRect.left > -1 ||
+						dropdownMenuRect.right < windowWidth
+					);
+				}
+			});
```

Comment:
Seeing a lot of code in `page.evaluate` makes me nervous :worried:  We should avoid this as much as possible, using PW standard API instead. If we don't, we're inviting a lot of strange bugs down the line.

Out of all of code I highlighted, I think the only line that must be in `page.evaluate` is setting of zoom, as this is [not yet implemented in PW.](https://github.com/microsoft/playwright/issues/2497). Everything else should be doable in PW. You can click on `Show Intermediate Pages` button, and check `isVisible()` on an item in dropdown that opened. Am I missing something here?

Reply:
@markocikos, I've tried using PW, after the zoom, for clicking on the Intermediate Pages button, but for some reason it's having a hard time clicking the button. 
This is the reason why I decided to move it into page.evaluate.

As for the dropdownVisible, as far as I know using isVisible, will not check if an element is clipping off the screen.

Let me know if there's any way around these or if I should try investigate further.
Thank you!

Reply:
Right, visible check is not correct. Did you try `toBeInViewport()`?

As for why button is not clicking, that sounds like typical PW flakiness, where you try to click before the button is ready for click event. To quickly confirm this you can try waiting for some fixed timeout, but better final solution, for that case, is to wait for some part in UI that indicates that button is ready.

Reply:
It looks like the button is not clicking because other elements are intercepting the click due to the view size being so small things are overlapping. I'm not too sure how to avoid this with OOTB PW.

I was able to use PW to check  ToBeInViewport() and it works, now it is just the button click not working.

Reply:
@fortunatomaldonado It looks like zoom is messing up PW. There might be a reason why they didn't implement it yet :slightly_smiling_face: 

I think we can just remove zoom change from the test. It's not required for the behavior we're fixing, it's just a method of achieving overflow, that we can also achieve with just the width. I tested this locally and this seems to do the job:
```js
await page.setViewportSize({height: 600, width: 200});

await page.getByTitle('Show Intermediate Pages').click();

await expect(page.getByLabel('Page 4')).toBeInViewport();
```

Note that I also reduced width to 200, so we get the break of pagination to new row. With 300, and no zoom, there was no break.

![image](https://github.com/user-attachments/assets/4900e136-6062-4d44-8882-9ffe37c27a52)